### PR TITLE
PMM-10503 backup modal

### PR DESCRIPTION
**Pull request by fabio-silva**
**What this PR does / why we need it**: Adds a checkbox to backup modal to change between on-demand / scheduled view

**Which issue(s) this PR fixes**: https://jira.percona.com/browse/PMM-10503

### CODE REVIEW

**Review by filipmikes1 — CHANGES_REQUESTED** [review]

File: public/app/percona/backup/components/AddBackupModal/AddBackupModal.utils.ts [hunk 1]
```diff
@@ -38,7 +38,10 @@ export const PERIOD_OPTIONS: Array<SelectableValue<PeriodType>> = [
 const isScheduledBackup = (backup: Backup | ScheduledBackup): backup is ScheduledBackup =>
   (backup as ScheduledBackup).cronExpression !== undefined;
 
-export const toFormBackup = (backup: Backup | ScheduledBackup | null): AddBackupFormProps => {
+const getBackupType = (backup: Backup | ScheduledBackup | null): BackupType =>
+  backup ? (isScheduledBackup(backup) ? BackupType.SCHEDULED : BackupType.DEMAND) : BackupType.DEMAND;
+
```
[the remark this answers is not in the document]
Comment (fabio-silva) [reply public/app/percona/backup/components/AddBackupModal/AddBackupModal.utils.ts:48]: Good one!

File: public/app/percona/backup/components/AddBackupModal/AddBackupModal.tsx [hunk 2]
```diff
@@ -80,6 +84,38 @@ export const AddBackupModal: FC<AddBackupModalProps> = ({
...
+                  {({ input }) => (
+                    <label>
+                      <input
+                        {...input}
+                        onChange={(e) => {
+                          setModalTitle(Messages.getModalTitle(false, editing));
+                          input.onChange(e);
+                        }}
+                      ></input>
+                      On Demand
```
[the remark this answers is not in the document]
Comment (fabio-silva) [reply public/app/percona/backup/components/AddBackupModal/AddBackupModal.tsx]: 👍

File: public/app/percona/backup/components/AddBackupModal/AddBackupModal.tsx [hunk 3]
```diff
@@ -80,6 +84,38 @@ export const AddBackupModal: FC<AddBackupModalProps> = ({
...
+                  {({ input }) => (
+                    <label>
+                      <input
+                        {...input}
+                        onChange={(e) => {
+                          setModalTitle(Messages.getModalTitle(true, editing));
+                          input.onChange(e);
+                        }}
+                      ></input>
+                      Schedule Backup
```
[the remark this answers is not in the document]
Comment (fabio-silva) [reply public/app/percona/backup/components/AddBackupModal/AddBackupModal.tsx]: 👍

**Review by filipmikes1 — APPROVED** [review]

**Review by filipmikes1 — DISMISSED** [review]

**Review by solovevayaroslavna — DISMISSED** [review]

**Review by filipmikes1 — APPROVED** [review]

**Comment by solovevayaroslavna** [comment]
Is it relevant? Can we close it? I have to much old PRs in my review list

**Pull request closed without merging**